src/user-activity/user-activity.service.ts

```diff
@@ -0,0 +1,157 @@
+import { UsersService } from 'src/users/users.service';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { actionType, UserActivity } from './entities/userActivity.entity';
+import UserActivitySave from './dto/user-activity-save.input';
+import { User } from 'src/users/entities/user.entity';
+import { Event } from 'src/events/entities/event.entity';
+@Injectable()
+export class UserActivityService {
+  constructor(
+    @InjectRepository(UserActivity)
+    private readonly userActivityRepository: Repository<UserActivity>,
+    private readonly usersService: UsersService,
+  ) {}
+
+  saveParticipantActivity(type, user, event) {
+    //Interested
+    let action = actionType.Interested;
+    let score = 2;
+    let weight = 0.03;
+    if (type == 2) {
+      //Take part
+      action = actionType.TakePart;
+      score = 3;
+      weight = 0.05;
+    }
+    this.createOrUpdate(user, action, event.type, score, weight);
+
+    this.createOrUpdate(user, actionType.Category, event.type, 1, 0.03);
+
+    this.createOrUpdate(
+      user,
+      actionType.Duration,
+      event.type,
+      this.checkDateScore(event),
+      0.02,
+    );
+  }
+
+  checkDateScore(event: any): number {
+    const startDate = new Date(event.startDate);
+    const endDate = new Date(event.endDate);
+    const diff = endDate.getTime() - startDate.getTime();
+    const hours = Math.floor(diff / 1000 / 60 / 60);
+    if (hours < 3) return 1;
+    if (hours > 7) return 3;
+
+    return 2;
+  }
+
+  async createOrUpdate(
+    user: any,
```

Comment:
Wrzućmy to w jeden argument który będzie obiektem z tymi wszystkimi właściwościami, tak jak pisałem wcześniej. Łatwiej się będzie podawać argumenty. Niewymagane argumenty w TypeScripcie możesz oznaczyć przez `?`